### PR TITLE
Add table_aws_ecs_cluster. Closes #136

### LINKED ISSUE #136: Add table aws_ecs_cluster
**Describe the new table**
An Amazon ECS cluster is a logical grouping of tasks or services

**References**
https://pkg.go.dev/github.com/aws/aws-sdk-go-v2/service/ecs

### PR DESCRIPTION
**Pull request by khushboo9024**
```TEST: tests/aws_ecs_cluster
Running terraform
data.aws_region.alternate: Refreshing state...
data.aws_partition.current: Refreshing state...
data.aws_region.primary: Refreshing state...
data.aws_caller_identity.current: Refreshing state...
data.null_data_source.resource: Refreshing state...
aws_ecs_cluster.named_test_resource: Creating...
aws_ecs_cluster.named_test_resource: Still creating... [10s elapsed]
aws_ecs_cluster.named_test_resource: Creation complete after 14s [id=arn:aws:ecs:us-east-1:013122550996:cluster/turbottest59015]

Warning: Deprecated Resource

The null_data_source was historically used to construct intermediate values to
re-use elsewhere in configuration, the same can now be achieved using locals


Apply complete! Resources: 1 added, 0 changed, 0 destroyed.

Outputs:

account_id = 013122550996
aws_partition = aws
region_name = us-east-1
resource_aka = arn:aws:ecs:us-east-1:013122550996:cluster/turbottest59015
resource_id = arn:aws:ecs:us-east-1:013122550996:cluster/turbottest59015
resource_name = turbottest59015

Running SQL query: test-get-query.sql
[
  {
    "active_services_count": 0,
    "cluster_arn": "arn:aws:ecs:us-east-1:013122550996:cluster/turbottest59015",
    "cluster_name": "turbottest59015",
    "status": "ACTIVE"
  }
]
✔ PASSED

Running SQL query: test-hydrate-query.sql
[
  {
    "akas": [
      "arn:aws:ecs:us-east-1:013122550996:cluster/turbottest59015"
    ],
    "cluster_arn": "arn:aws:ecs:us-east-1:013122550996:cluster/turbottest59015",
    "cluster_name": "turbottest59015",
    "tags": {
      "name": "turbottest59015"
    },
    "title": "turbottest59015"
  }
]
✔ PASSED

Running SQL query: test-list-query.sql
[
  {
    "active_services_count": 0,
    "attachments": "<null>",
    "attachments_status": "<null>",
    "capacity_providers": [],
    "cluster_arn": "arn:aws:ecs:us-east-1:013122550996:cluster/turbottest59015",
    "cluster_name": "turbottest59015",
    "default_capacity_provider_strategy": [],
    "pending_tasks_count": 0,
    "registered_container_instances_count": 0,
    "running_tasks_count": 0,
    "settings": [
      {
        "Name": "containerInsights",
        "Value": "disabled"
      }
    ],
    "statistics": [],
    "status": "ACTIVE"
  }
]
✔ PASSED

Running SQL query: test-not-found-query.sql
null
✔ PASSED

Running SQL query: test-turbot-query.sql
[
  {
    "account_id": "013122550996",
    "akas": [
      "arn:aws:ecs:us-east-1:013122550996:cluster/turbottest59015"
    ],
    "cluster_arn": "arn:aws:ecs:us-east-1:013122550996:cluster/turbottest59015",
    "cluster_name": "turbottest59015",
    "region": "us-east-1",
    "title": "turbottest59015"
  }
]
✔ PASSED

POSTTEST: tests/aws_ecs_cluster

TEARDOWN: tests/aws_ecs_cluster

SUMMARY:

1/1 passed.```

### CODE REVIEW

File: aws/table_aws_ecs_cluster.go [hunk 1]
```diff
@@ -0,0 +1,283 @@
...
+				Description: "The Amazon Resource Name (ARN) that identifies the cluster.",
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromField("ClusterArn"),
+			},
+			{
+				Name:        "cluster_name",
+				Description: "A user-generated string that you use to identify your cluster.",
+				Type:        proto.ColumnType_STRING,
+				Hydrate:     getEcsCluster,
+				Transform:   transform.FromField("ClusterName"),
```
Comment (Subhajit97) [inline aws/table_aws_ecs_cluster.go:41]:
```suggestion
```

File: aws/table_aws_ecs_cluster.go [hunk 2]
```diff
@@ -0,0 +1,283 @@
...
+			},
+			{
```
Comment (Subhajit97) [inline aws/table_aws_ecs_cluster.go:43]: Don't need to add `transform.FromField()` if the camelCase value of column name is same as SDK property.

File: aws/table_aws_ecs_cluster.go [hunk 3]
```diff
@@ -0,0 +1,283 @@
...
+
+	return clusterdata, nil
+}
+
+//// TRANSFORM FUNCTIONS 
+
+func getAwsEcsClusterTurbotTags(ctx context.Context, d *transform.TransformData) (interface{},
+error) {
+	ecsClusterTags := d.HydrateItem.(*ecs.ListTagsForResourceOutput)
+
```
Comment (Subhajit97) [inline aws/table_aws_ecs_cluster.go:274]: Return `nil`, if the `ecsClusterTags.Tags` return nil
```suggestion

if ecsClusterTags.Tags == nil {
  return nil, nil
}

```

File: docs/tables/aws_ecs_cluster.md [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+  cluster_arn,
+  status
+from
+  aws_ecs_cluster
+where
+  status IN ('FAILED','INACTIVE');
+```
+
+
+### List resources attached to ecs clusters
```
Comment (Subhajit97) [inline docs/tables/aws_ecs_cluster.md:42]:
```suggestion
### List of resources attached with clusters
```

File: docs/tables/aws_ecs_cluster.md [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+
+```sql
+select
+  attachments,
+  attachments_status
+from
+  aws_ecs_cluster;
```
Comment (Subhajit97) [inline docs/tables/aws_ecs_cluster.md:49]: Can we add some specific details for attachements (attachment_type, attachment_status, attachement_id)?

File: docs/tables/aws_ecs_cluster.md [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+
+### List of cluster for which CloudWatch Container Insights is disabled
+
+```sql
+select
+  cluster_arn,
+  setting ->> 'Name' as Name,
+  setting ->> 'Value' as Value
+from
+  aws_new.aws_ecs_cluster,
```
Comment (Subhajit97) [inline docs/tables/aws_ecs_cluster.md:61]:
```suggestion
  aws_ecs_cluster,
```

File: aws-test/tests/aws_ecs_cluster/variables.tf [hunk 7]
```diff
@@ -0,0 +1,84 @@
...
+resource "aws_ecs_cluster" "named_test_resource" {
+  name = var.resource_name
+  tags = {
+    name = var.resource_name
+  }
+}
+
+# Get the Turbot resource that represents the test resource. This shadow will be used
+# for all policy settings etc throughout the test. Waiting for the shadow also ensures
+# proper ordering of the setup of policies etc after the resource exists in Turbot.
```
Comment (Subhajit97) [inline aws-test/tests/aws_ecs_cluster/variables.tf:60]:
```suggestion
```

**Review by Subhajit97 — CHANGES_REQUESTED** [review]
Make suggested changes, and add the tests logs

File: aws/table_aws_ecs_cluster.go [hunk 8]
```diff
@@ -0,0 +1,283 @@
...
+	pagesLeft := true
+
+	for pagesLeft {
+		result, err := svc.ListClusters(params)
+		if err != nil {
+			return nil, err
+		}
+
+		for _, results := range result.ClusterArns {
+			plugin.Logger(ctx).Trace("clusterArnclusterArnclusterArn", "clusterArn", results)
```
Comment (Subhajit97) [inline aws/table_aws_ecs_cluster.go:183]:
```suggestion
```

File: aws/table_aws_ecs_cluster.go [hunk 9]
```diff
@@ -0,0 +1,267 @@
...
+				Description: "The settings for the cluster. This parameter indicates whether CloudWatch Container Insights is enabled or disabled for a cluster.",
+				Type:        proto.ColumnType_JSON,
+				Hydrate:     getEcsCluster,
+			},
+			{
+				Name:        "statistics",
+				Description: "Additional information about your clusters that are separated by launch type.",
+				Type:        proto.ColumnType_JSON,
+				Hydrate:     getEcsCluster,
+				Transform:   transform.FromField("Statistics"),
```
Comment (Subhajit97) [inline aws/table_aws_ecs_cluster.go:94]:
```suggestion
```

**Review by Subhajit97 — DISMISSED** [review]
LGTM

**Review by cbruno10 — APPROVED** [review]

**Pull request merged**